### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by angrave**
staging to master

### CODE REVIEW

File: ClassTranscribeDatabase/Globals.cs [hunk 1]
```diff
@@ -17,6 +18,9 @@ public class AppSettings
         public string ADMIN_USER_ID { get; set; }
         public string ADMIN_PASSWORD { get; set; }
         public string RabbitMQServer { get; set; }
+
```
Comment (robkooper) [inline ClassTranscribeDatabase/Globals.cs:21]: we should be consistent with variables like this. I think it be good to look at a linter that checks formatting (yes I care about that).

See 
- https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions
- https://google.github.io/styleguide/csharp-style.html

Might be interesting to look at:
https://marketplace.visualstudio.com/items?itemName=ChrisDahlberg.StyleCop

**Review by robkooper — APPROVED** [review]
some generic comments.

File: CTCommons/RabbitMQ/RabbitMQConnection.cs [hunk 2]
```diff
@@ -32,12 +46,25 @@ public RabbitMQConnection(ILogger<RabbitMQConnection> logger)
             {
                 HostName = Globals.appSettings.RabbitMQServer,
                 UserName = Globals.appSettings.ADMIN_USER_ID,
-                Password = Globals.appSettings.ADMIN_PASSWORD
+                Password = Globals.appSettings.ADMIN_PASSWORD,
+                Port = Convert.ToUInt16(Globals.appSettings.RABBITMQ_PORT) // 5672
+
             };
+            _logger.LogInformation($"Connection to RabbitMQ server {factory.HostName} with user {factory.UserName} on port {factory.Port}...");
```
[the remark this answers is not in the document]
Comment (angrave) [reply CTCommons/RabbitMQ/RabbitMQConnection.cs:53]: Made a new issue--
https://github.com/classtranscribe/WebAPI/issues/104

File: CTCommons/RabbitMQ/RabbitMQConnection.cs [hunk 3]
```diff
@@ -90,24 +122,34 @@ public void ConsumeTask<T>(string queueName, Func<T, TaskParameters, Task> OnCon
...
                 _channel.BasicQos(prefetchSize: 0, prefetchCount: concurrency, global: false);
             }
-            
+
             _logger.LogInformation(" [*] Waiting for messages, queueName - {0}", queueName);
 
             var consumer = new EventingBasicConsumer(_channel);
             consumer.Received += async (model, ea) =>
             {
+                // This object exists so that we can wrap all OnConsumes with a try-finally here
```
[the remark this answers is not in the document]
Comment (angrave) [reply CTCommons/RabbitMQ/RabbitMQConnection.cs:131]: So what do you recommend?

File: CTCommons/MSTranscription/KeyProvider.cs [hunk 4]
```diff
@@ -35,15 +35,26 @@ public KeyProvider(AppSettings appSettings)
...
             foreach (string subscriptionKey in subscriptionKeys.Split(';'))
             {
+                if (!subscriptionKeys.Contains(","))
+                {
+                    continue;
+                }
+                string[] keyregion = subscriptionKey.Split(',');
+                if (keyregion.Length != 2)
+                {
+                    throw new Exception("AZURE_SUBSCRIPTION_KEYS should be in the form key,region;key,region");
```
[the remark this answers is not in the document]
Comment (angrave) [reply CTCommons/MSTranscription/KeyProvider.cs:48]: There was on logger in this class! Will fix though in a future commit

File: CTCommons/RabbitMQ/RabbitMQConnection.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (robkooper) [reply CTCommons/RabbitMQ/RabbitMQConnection.cs:131]: what we did was complex, not sure it is the best yet:

```
worker = None

while(consumeMessage(timeout = 100ms)) {
  if worker and worker.finished:
     channel.ack
  if newjob and worker = None
     worker = new Job(finished=False)
}

Job() {
  finished = False
  run():
     do some work
     finished=True
```

**Pull request merged**